src/vs/platform/actionWidget/browser/actionList.ts

```diff
@@ -302,7 +303,23 @@ export class ActionList<T> extends Disposable {
 		}
 	}
 
-	private onListHover(e: IListMouseEvent<IActionListItem<T>>): void {
+	private async onListHover(e: IListMouseEvent<IActionListItem<T>>): Promise<void> {
+
+		const element = e.element;
+		if (element) {
+			if (element.item && this.focusCondition(element)) {
+				if (this._delegate.onFocus) {
+					const result = await this._delegate.onFocus(element.item);
+					element.canPreview = result.canPreview;
+				}
+				if (e.index) {
+					this._list.splice(e.index, 1, [element]);
```

Comment:
Do you need both the `splice` and the call to `rerender`? Or is `rerender` enough